api/v1alpha1/nnf_node_storage_types.go

```diff
@@ -37,6 +37,15 @@ type NnfNodeStorageSpec struct {
 	// +kubebuilder:validation:Minimum:=0
 	Count int `json:"count"`
 
+	// User ID for file system
+	UserID uint32 `json:"userID"`
+
+	// Group ID for file system
+	GroupID uint32 `json:"groupID"`
+
+	// +kubebuilder:default:=false
+	SetOwnerGroup bool `json:"setOwnerGroup"`
```

Comment:
Why wouldn't we always set the owner/group? Thus, you wouldn't need this.

Reply:
It's only for Lustre file systems, and it only gets set for one of the nnfNodeStorage resources. The Lustre file system has lots of different Lustre targets (MDTs and OSTs) each with an nnfNodeStorage, but we only need to have one client to mount the file system and set the permissions. OST 0 is always picked as the nnfNodeStorage to make "SetOwnerGroup".